contributing/deprecation_policy.md

```diff
@@ -4,27 +4,89 @@ The ability to deprecate features, components and paradigms is essential to the
 
-## Get the 'thumbs up'
+Before embarking on the deprecation of any API we should collaboratively answer the following questions:
 
-Propose your deprecation on GitHub as an issue where the title's first phrase is "Deprecation [Component Name]". Much like a feature proposal, the contributors have a responsibility to carefully judge the consequences of breaking changes against possible gains. 
-
-We should collaboratively answer the following questions:
-
-1. Should this code be deprecated? Is it really worth the time and effort for everyone?
+1. Should this code be deprecated? Is the effort required to deprecate this component worth the benefit of removing the old behavior?
 1. Why is the code being deprecated? Why should clients switch?
 1. What are clients supposed to replace the code with?
 1. What is a good deprecation timeline?
-    1. Small changes can be done in a month.
-    1. Larger changes can be done in a quarter/three months.
-    1. Adjust up or down based on the size/complexity of the change.
-1. Who’s on point? It’s good to have a single person to keep track of things.
+1. Who is responsible for the deprecation plan?
+
+## Our deprecation process
+
+1. Measure usage of the API.
+2. Create a GitHub project.
+3. Write a migration guide.
+4. Annotate the API as "ToBeDeprecated".
+5. Determine a schedule for the API deletion.
+6. Announce that the API will be deprecated
+7. After the deprecation date, deprecate the API.
+8. After the deletion date, re-evaluate usage of the API.
+9. Delete the API.
+
+### Step 1: Measure usage of the API
+
+Read [go/mdc-ios-measuring-api-usage](http://go/mdc-ios-measuring-api-usage) to learn how to measure API usage internally.
+
+If there is zero API usage beyond our own library: skip to step 9 (Delete the API).
+Otherwise, proceed to step 2.
+
+### Step 2: Create a GitHub project
+
+Visit [go/mdc-ios-project-template](http://go/mdc-ios-project-template) or this [GitHub search](https://github.com/material-components/material-components-ios/projects?utf8=%E2%9C%93&query=is%3Aopen+Template+project) to access the template GitHub project.
+
+Name the project "Deprecate YYY". Use this project to organize any work related to this deprecation.
+
+Create at least one issue in the project titled "Delete YYY". This will be the last issue you close once the API is deleted.
+
+### Step 3: Write a migration guide
+
+This guide should be placed in the component's docs/ folder and have a prefix `migration-guide-`.
+
+For example, `migration-guide-appbar-appbarviewcontroller.md` is a guide from MDCAppBar to MDCAppBarViewController.
+
+Send this migration guide out for review as a standalone pull request.
+
+### Step 4: Annotate the API as "ToBeDeprecated"
+
+Add a `@warning` annotation to the API's public documentation indicating the intent for this API to eventually be deprecated. This warning should also provide a short suggested alternative, if one exists, and a link to the migration guide.
```

Comment:
Nit: formatting/line length for this file.

Reply:
Done.